components/online_image.rst

```diff
@@ -0,0 +1,134 @@
+.. _online_image:
+
+Online Image Component
+======================
+
+.. seo::
+    :description: Instructions for displaying images downloaded at runtime in ESPHome.
+    :image: image-sync-outline.svg
+
+With this component you can define images that will be downloaded, decoded and drawn at runtime.
+
+.. note::
+
+    Currently only images in PNG format are supported.
+
+.. warning::
+
+    This component requires a fair amount of RAM; both for downloading the image, and for storing the decoded image. It might work on devices without PSRAM, but there is no guarantee.
+
+This component has a dependency to :doc:`/components/http_request`; the configuration options you set to the ``http_request`` component will also apply here.
+
+.. code-block:: yaml
+
+    http_request:
+
+    online_image:
+      - url: "https://upload.wikimedia.org/wikipedia/commons/thumb/4/47/PNG_transparency_demonstration_1.png/280px-PNG_transparency_demonstration_1.png"
+        format: png
+        id: my_online_image
+        on_download_finished:
+          component.update: my_display
+
+And then later in code:
+
+.. code-block:: yaml
+
+    display:
+      - platform: ...
+        id: my_display
+        # ...
+        lambda: |-
+          // Draw the image my_online_image at position [x=0,y=0]
+          it.image(0, 0, id(my_online_image));
+
+For monochrome displays the ``image`` method accepts two additional color parameters which can
+be supplied to specify the color used to draw bright and dark pixels respectively.
+In this case the image will be internally converted to a grayscale image and then to monochrome
+based on an internally defined threshold.
+
+.. code-block:: yaml
+
+    display:
+      - platform: ...
+        id: my_display
+        # ...
+        lambda: |-
+          // Draw the image my_image at position [x=0,y=0]
+          // with front color "OFF" and back color "ON"
+          it.image(0, 0, id(my_online_image), COLOR_OFF, COLOR_ON);
+
+By default ``online_image`` is configured to not automatically update/download the image; in order to do the initial download, you can either:
+ - Add a ``component.update <image_id>`` in the ``on_connect:`` action on the :doc:`/components/wifi` component.
+ - Explicitly set an ``update_interval``.
+ - Call ``component.update <image_id>`` in an :doc:`/components/interval` block.
+ - Call ``component.update <image_id>`` where you need the image to be downloaded/updated.
+
+.. code-block:: yaml
+
+    wifi:
+      on_connect:
+        - component.update: my_online_image
```

Comment:
Minimal config example only first please. A full example is allowed to go below the `Actions` section